eth/protocols/snap/sync.go

```diff
@@ -758,6 +782,17 @@ func (s *Syncer) loadSyncStatus() {
 						options = options.WithWriter(func(path []byte, hash common.Hash, blob []byte) {
 							rawdb.WriteTrieNode(subtask.genBatch, owner, path, hash, blob, s.scheme)
 						})
+						if s.scheme == rawdb.PathScheme {
+							// Configure the dangling node cleaner and also filter out boundary nodes
+							// only in the context of the path scheme. Deletion is forbidden in the
+							// hash scheme, as it can disrupt state completeness.
+							options = options.WithCleaner(func(path []byte) {
+								s.cleanPath(subtask.genBatch, owner, path)
+							})
+							options = options.SkipBoundary(true, true, func(path []byte, hash common.Hash, blob []byte) {
```

Comment:
The `cleaner` should only be used in path mode, but the `SkipBoundary` might aswell used regardless, no?